src/tir/transforms/storage_access.cc

```diff
@@ -94,6 +98,20 @@ void StorageAccessVisitor::VisitStmt_(const EvaluateNode* op) {
   allow_append_ = false;
 }
 
+void StorageAccessVisitor::VisitStmt_(const LetStmtNode* op) {
+  allow_append_ = true;
+  curr_stmt_.access.clear();
```

Comment:
shouldn't call clear here, it can only be used after finishing handling a statement

Reply:
can you check my new solution?